Dockerfile

```diff
@@ -5,41 +5,11 @@ RUN rustup component add rustfmt
 
 WORKDIR /usr/src
 
-# Use a dummy entrypoint to build each crate, that way dependencies can be cached
-# without code changes causing a complete rebuild.
-# TODO all these COPY commands add additional layers to the image. I'm not sure
-# how big they are but for the distributed images we likely want to use
-# a simpler Dockerfile.
-COPY Cargo.toml .
-RUN mkdir -p pcli/src pd/src &&\
-    echo "fn main() {}" > pcli/src/main.rs &&\
-    echo "fn main() {}" > pd/src/main.rs
-COPY pcli/build.rs ./pcli
-COPY pcli/Cargo.toml ./pcli
-COPY pd/Cargo.toml ./pd
-COPY pd/build.rs ./pd
-# TODO If the protobuf definitions, crypto, or wallet implementations change,
-# there will be a complete rebuild. This is maybe possible to avoid.
-COPY proto ./proto
-COPY crypto ./crypto
-COPY decaf377-fmd ./decaf377-fmd
-COPY decaf377-ka ./decaf377-ka
-COPY wallet ./wallet
-COPY .git ./.git
-# Sorry about all that mess ^, but it's worth it during development.
+COPY . .
```

Comment:
How does this relate to the COPY commands for all the caching layers in the non-release image? (I mean, I don't understand what `COPY . .` is doing)

Reply:
The `COPY . .` copies the entirety of the working directory on the host machine into the Docker build container, i.e. copies the full source tree in. The previous statements just copied over the `Cargo.toml` files to get the dependencies and built empty dummy `pd`/`pcli` implementations to be fast.

Reply:
Oh, is this running the `Dockerfile.dev` first and then loading it in here or something?